### PR TITLE
 EPMRPP-35576/EPMRPP-35565 React. Launches. Post issue modal (also on the log level)

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx [hunk 1]
```diff
@@ -98,11 +88,8 @@ export class LinkIssueModal extends Component {
 
   constructor(props) {
     super(props);
-    this.pluginNamesOptions = Object.keys(props.namedBtsIntegrations).map((key) => ({
-      value: key,
-      label: INTEGRATION_NAMES_TITLES[key],
-    }));
-    const pluginName = this.pluginNamesOptions[0].value;
+    const pluginName = Object.keys(props.namedBtsIntegrations)[0];
+    console.log(props.namedBtsIntegrations);
```
Comment (mykhailo-kalashnikov-epam) [inline app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx]: Remove console.log

Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx]: Fixed

File: app/src/pages/inside/stepPage/modals/postIssueModal/utils.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    validValues = Object.assign(validValues, validatedFields);
+  }
+  return validValues;
+};
+
+export const createFieldsValidationConfig = (fields) =>
+  fields.reduce((acc, item) => ({ ...acc, [item.id]: item.required }), {});
+
+export function getSessionStorageItem(key) {
```
Comment (DiscoElevator) [inline app/src/pages/inside/stepPage/modals/postIssueModal/utils.js]: Why do we need session storage?

File: app/src/pages/inside/stepPage/modals/postIssueModal/constants.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export const LOG_QUANTITY = 50;
+export const SAVED_BTS_CREDENTIALS_KEY = 'btsIntegrationCredentials';
+
+export const DEFAULT_INCLUDE_DATA_CONFIG = {
+  [INCLUDE_ATTACHMENTS_KEY]: true,
+  [INCLUDE_LOGS_KEY]: true,
+  [INCLUDE_COMMENTS_KEY]: true,
+};
+
+export const BULK_INCLUDE_DATA_CONFIG = {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/constants.js]: In the initial values. I rewrote it with function.

[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/constants.js]: In the initial values. I rewrote it with function.

File: app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+    };
+  }
+
+  onPost = () => (closeModal) => {
+    this.closeModal = closeModal;
+    this.props.handleSubmit(this.prepareDataToSend)();
+  };
+
+  onChangePlugin = (pluginName) => {
+    if (pluginName !== this.state.pluginName) {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx]: Fixed

File: app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+      this.setState({
+        pluginName,
+        fields,
+        integrationId: id,
+      });
+    }
+  };
+
+  onChangeIntegration = (integrationId) => {
+    if (integrationId !== this.state.integrationId) {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx]: Fixed

File: app/src/pages/inside/stepPage/modals/postIssueModal/utils.js (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/utils.js]: We discuss it earlier but I don't have any reasons nowadays. Replaced with `localStorage`

File: app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+  };
+
+  prepareDataToSend = (formData) => {
+    const {
+      data: { items },
+    } = this.props;
+
+    const fields = this.state.fields.map((field) => ({ ...field, value: formData[field.id] }));
+    const backLinks = items.reduce(
+      (acc, item) => ({ ...acc, [item.id]: `${window.location.toString()}/${item.id}/log` }),
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/stepPage/modals/postIssueModal/postIssueModal.jsx]: Fixed. Added proper selector to `controllers/testItem/selectors.js`.

**Pull request merged**